lib/signaling/v3/remoteparticipant.js

```diff
@@ -0,0 +1,66 @@
+'use strict';
+
+const RemoteParticipantV2 = require('../v2/remoteparticipant');
+const RemoteTrackPublicationV3 = require('./remotetrackpublication');
+
+/**
+ * @extends RemoteParticipantV2
+ */
+class RemoteParticipantV3 extends RemoteParticipantV2 {
+  /**
+   * Construct a {@link RemoteParticipantV2}.
+   * @param {object} participantState
+   * @param {function(Track.SID): Promise<MediaTrackReceiver>} getPendingTrackReceiver
+   * @param {function(Track.SID): boolean} getInitialTrackSwitchOffState
+   * @param {function(Track.SID, Track.Priority): boolean} setPriority
+   * @param {function(Track.SID, ClientRenderHint): Promise<void>} setRenderHint
+   * @param {function(Track.SID): void} clearTrackHint
+   * @param {object} [options]
+   */
+  constructor(
+    participantState,
+    getPendingTrackReceiver,
+    getInitialTrackSwitchOffState,
+    setPriority,
+    setRenderHint,
+    clearTrackHint,
+    options
+  ) {
+    options = Object.assign({
+      RemoteTrackPublicationSignaling: RemoteTrackPublicationV3,
+      otherProperties: {
+        _getPendingTrackReceiver: getPendingTrackReceiver
+      }
+    }, options);
+
+    super(
+      participantState,
+      getInitialTrackSwitchOffState,
+      setPriority,
+      setRenderHint,
+      clearTrackHint,
+      options
+    );
+  }
+
+  /**
+   * @private
+   */
+  _getOrCreateTrack(trackState) {
+    const {
+      _RemoteTrackPublicationSignaling: RemoteTrackPublicationV3,
```

Comment:
maybe it's worth keeping both `RemoteTrackPublicationV3` and `RemoteTrackPublicationV2` in options object (instead of conditional _RemoteTrackPublicationSignaling), so that we do not need to rename it here.

Reply:
I don't think we should do that. In principle, if there is not much difference between v2 and v3, then at the base class level, a more generic name is much more descriptive of what is going on.